spec/fabricators/application_failure_fabricator.rb

```diff
@@ -0,0 +1,12 @@
+# Simulates Temporal::Connection::Serializer::Failure
+Fabricator(:application_failure, from: Temporal::Api::Failure::V1::Failure) do
+  transient :error_class, :backtrace
+  message { |attrs| attrs[:message] }
+  stack_trace { |attrs| attrs[:backtrace].join("\n") }
+  application_failure_info do |attrs|
+    Temporal::Api::Failure::V1::ApplicationFailureInfo.new(
+      type: attrs[:error_class],
+      details: TestDeserializer.new.to_details_payloads(attrs[:message]),
```

Comment:
This will fail when used in any other spec since `TestDeserializer` is only defined in a your new specs

Reply:
Also since the fabricator is for a proto message it probably should not be concerned with error serialisation, but passed details for the error directly

Reply:
Good flag.  As for the second suggestion, feels like reusing production methods rather than reimplementing them trumps strict dependencies here.